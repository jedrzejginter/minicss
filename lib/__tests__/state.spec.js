const { mapState, statesMapObject } = require('../state');

describe('statesMapObject', () => {
  test('should remain unchanged', () => {
    expect(statesMapObject).toMatchSnapshot();
  });
});

describe('mapState', () => {
  it('should fall back to "default"', () => {
    expect(mapState('')).toBe('default');
  });

  it('should correctly recognize "a" as "active"', () => {
    expect(mapState('a')).toBe('active');
  });

  it('should correctly recognize "c" as "checked"', () => {
    expect(mapState('c')).toBe('checked');
  });

  it('should correctly recognize "d" as "disabled"', () => {
    expect(mapState('d')).toBe('disabled');
  });

  it('should correctly recognize "e" as "empty"', () => {
    expect(mapState('e')).toBe('empty');
  });

  it('should correctly recognize "f" as "focus"', () => {
    expect(mapState('f')).toBe('focus');
  });

  it('should correctly recognize "h" as "hover"', () => {
    expect(mapState('h')).toBe('hover');
  });

  it('should correctly recognize "v" as "visited"', () => {
    expect(mapState('v')).toBe('visited');
  });
});
