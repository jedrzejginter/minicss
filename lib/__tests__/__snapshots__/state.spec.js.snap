// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`statesMapObject should remain unchanged 1`] = `
Object {
  "a": "active",
  "c": "checked",
  "d": "disabled",
  "e": "empty",
  "f": "focus",
  "h": "hover",
  "v": "visited",
}
`;
